doc/development/powershell.md

```diff
@@ -0,0 +1,165 @@
+- Example pester test suites: [job matrix tests](https://github.com/Azure/azure-sdk-tools/tree/main/eng/common/scripts/job-matrix/tests), [asset sync tests](https://github.com/Azure/azure-sdk-tools/blob/main/tools/asset-sync/assets.Tests.ps1)
+- A CI pipeline should be defined to run scripts unit tests at the very least. See [archetype-sdk-tool-pwsh](https://github.com/Azure/azure-sdk-tools/blob/main/eng/pipelines/templates/stages/archetype-sdk-tool-pwsh.yml) for how to do this.
+- Script code should always be written so as much of the surface area as possible can be run via unit tests. Move code that calls out to external dependencies into modular functions, and simplify context/data structures passed to functions as much as possible to it can be easily mocked.
+
+#### Running Pester Tests
+
+(stolen from https://github.com/Azure/azure-sdk-tools/blob/main/tools/asset-sync/contributing.md).
+
+> **First, ensure you have `pester` installed:**
+>
+> `Install-Module Pester -Force`
+> 
+> Then invoke tests with:
+> 
+> `Invoke-Pester ./assets.Tests.ps1`
+> 
+> **See stdout output**
+> 
+> `Invoke-Pester <other arguments> -PassThru`
+> 
+> **Select a subset**
+> 
+> To select a subset of tests, invoke using the `FullNameFilter` argument. EG
+> 
+> `Invoke-Pester ./assets.Tests.ps1 -FullNameFilter="*Evaluate-Target-Dir*"`
+> 
+> The "full-name" is simply the full namespace including parent "context" names.
+> 
+> ```powershell
+> Describe "AssetsModuleTests" {
+>   Context "Evaluate-Target-Dir" {
+>     It "Should evaluate a root directory properly." {
+>     }
+>   }
+> }
+> ```
+> 
+> Full evaluated test name is going to be:
+> 
+> `AssetsModuleTests.Evaluate-Target-Dir.Should evaluate a root directory properly.`
+
+### Local/Pipeline Functional Testing
+
+Some functionality relies so heavily on external process or service calls that unit tests do not provide a good return on investment. For these cases, we should still favor being able to run and validate the code as easily as possible.
+
+Some tips for writing testable scripts:
+
+- Avoid referencing implicit values that require environmental setup. For example, if a script processes an azure pipeline artifact, don't hardcode the standard azure pipeline artifact location, but rather add a parameter with that location as the default value.
+- Structure large scripts into multiple functions and/or files. This way you can test functions with smaller pieces of logic in isolation.
+  - A useful pattern to enable testing like this is to have two scripts: one "wrapper script" to define command line parameters and do any high level transformations, and another "library script" to contain all the functions and a main entrypoint function that the wrapper script can call. In order to test logic in isolation from the library script, support dot sourcing:
+    - Add this block to the bottom of your library script.
+      ```
+      if ($MyInvocation.InvocationName -ne ".") {
```

Comment:
This doesn't follow the style you described above about braces on new lines, but I think - for consistency across scriptblocks and functions - you should use this style ubiquitously.